src/sparseml/transformers/utils/helpers.py

```diff
@@ -224,3 +207,92 @@ def resolve_sequence_length(config: AutoConfig) -> int:
         "(inferred from HF transformers config) "
     )
     return sequence_length
+
+
+def resolve_recipe_application(
+    recipe: Union[str, Path, None], model_path: Union[str, Path]
+) -> Union[str, Path, None]:
+    """
+    Resolve the recipe to apply to the model.
+    :param recipe: the recipe to apply to the model.
+        It can be one of the following:
+        - None (no recipe will be applied or the
+            default recipe will be applied if exists. Default recipe
+            is assumed to be stored in the model_path and named RECIPE_NAME)
+        - a path to the recipe file
+        - name of the recipe file (e.g. "recipe.yaml")
+            (assumed to be stored in the model_path instead
+            of RECIPE_NAME)
+        - a string containing the recipe
+    :param model_path: the path to the model to load
+    :return: the resolved recipe
+    """
+
+    if recipe is None:
+        # if recipe is None -> still look for recipe.yaml in the model_path
+        recipe = os.path.join(model_path, RECIPE_NAME)
+        if os.path.isfile(recipe):
+            return recipe
```

Comment:
could we also deal with the case where the model_path is a HF model_id, which loads fine with `SparseAutoModel` already, that has a `recipe.yaml` in it? 

I implemented an example of this in my earlier SparseAutoModel examples
```python
# Load model with SparseAutoModel
from sparseml.transformers.utils import SparseAutoModel
from transformers import AutoConfig

config = AutoConfig.from_pretrained(model_id)  # Why does SparseAutoModel need config?
model = SparseAutoModel.text_generation_from_pretrained(model_id, config=config)

# Apply recipe to model
from sparseml.pytorch.model_load.helpers import apply_recipe_structure_to_model
from huggingface_hub import hf_hub_download
import os

# vv This part is important vv
recipe_path = hf_hub_download(repo_id=model_id, filename="recipe.yaml")
apply_recipe_structure_to_model(model=model, recipe_path=recipe_path, model_path=os.path.dirname(recipe_path))
```

Reply:
Added the appropriate pathway and tested by running:
```python
from sparseml.transformers.utils.sparse_model import SparseAutoModelForCausalLM

SparseAutoModelForCausalLM.from_pretrained("mgoin/starcoderbase-1b-pruned50")
```

```bash
2024-01-26 13:06:41 sparseml.transformers.utils.helpers INFO     model_path is a huggingface model id. Attempting to download recipe from https://huggingface.co/
Logging all SparseML modifier-level logs to sparse_logs/26-01-2024_13.06.41.log
2024-01-26 13:06:41 sparseml.core.logger.logger INFO     Logging all SparseML modifier-level logs to sparse_logs/26-01-2024_13.06.41.log
2024-01-26 13:06:41 sparseml.core.recipe.recipe INFO     Loading recipe from file /home/damian/.cache/huggingface/hub/models--mgoin--starcoderbase-1b-pruned50/snapshots/07a616d5bd1e8f63b8564b116046ac44e79079de/recipe.yaml
manager stage: Model structure initialized
2024-01-26 13:06:41 sparseml.pytorch.model_load.helpers INFO     Applied an unstaged recipe to the model at mgoin/starcoderbase-1b-pruned50
2024-01-26 13:06:41 sparseml.pytorch.model_load.helpers WARNING  Model state was not reloaded for SparseML: could not find model weights for mgoin/starcoderbase-1b-pruned50
```

Reply:
What about the SparseZoo case?

Reply:
Good call. Adding it!